docs/workflow/trimming/feature-switches.md

```diff
@@ -20,6 +20,7 @@ configurations but their defaults might vary as any SDK can set the defaults dif
 | TBD | System.Threading.ThreadPool.EnableDispatchAutoreleasePool | When set to true, creates an NSAutoreleasePool around each thread pool work item on applicable platforms. |
 | CustomResourceTypesSupport | System.Resources.ResourceManager.AllowCustomResourceTypes | Use of custom resource types is disabled when set to false. ResourceManager code paths that use reflection for custom types can be trimmed. |
 | EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization | BinaryFormatter serialization support is trimmed when set to false. |
+| BuiltInComSupport | System.Runtime.InteropServices.Marshal.IsBuiltInComSupported | In-built COM support is trimmed when set to false. |
```

Comment:
I don't understand why this is named `BuiltInComSupport`. Isn't every feature "built in"?

Reply:
> System.Runtime.InteropServices.Marshal.IsBuiltInComSupported

I would suggest `System.Runtime.InteropServices.Com.IsSupported` to align with the naming pattern elsewhere - `Debugger.IsSupported`, `EventSoruce.IsSupported`, etc.

Reply:
"COM" will still be supported even with this feature switch off - but only through [`ComWrappers`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.comwrappers?view=net-5.0). This FS should disable the seamless COM which lot of people use today. It's basically similar to sourcegen (ComWrappers) and reflection (Built-in COM).

@AaronRobinsonMSFT Do we have a better name for the "built-in" COM?

Reply:
I rather we keep this name if possible since we went over this naming convention multiple times in the earlier feature checkin and this was the name that was suggested by at that time

Reply:
If we want to keep the terms "Built-in COM Support", I'm fine with that since those are the words used in the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.comwrappers?view=net-5.0) @vitek-karas linked above.

However, I'd still follow the established feature-switch naming pattern: `<namespace>.<feature-name>.IsSupported`. i.e. `System.Runtime.InteropServices.BuiltInCom.IsSupported`.

Reply:
This is a good question. From a runtime perspective this is split as `FEATURE_COMINTEROP` and `FEATURE_COMWRAPPERS`. If I were naming things I would call it `ComInterop`. I say that because `ComWrappers` provides nothing by default &ndash; it is a simple API/building block to enable COM interop with source generation. As a metaphor consider ECMA-335. It isn't a type system but it is a contract that enables building up a type system. That is how I view `ComWrappers` vs COM interop.

Reply:
I prefer this property to be part of the Marshal class since the semantics have to be be understood in non-trim scenarios as well and the trim naming doesn't obfuscate the general case too much. Ideally, we would like all use of built-in COM to be replaced by ComWrappers in the long term. I dont' see a consistent pattern in all the feature switch usage but will look to see if there is a way to be consistent with trim naming without causing too much confusion in the general case.

Reply:
A thing I want to make clear here is that the `AppContext Setting` name does NOT need to match the C# type name and property at all. The `AppContext Setting` name should be treated like public API - once we ship with this name, we will need to support that name going forward. Customers are going to write these strings into their .csproj/.targets/.json files, and changing the strings will break them.

I don't feel putting the `Marshal` name in this app context setting adds any value to the user. So I would drop it from the setting name. Depending on whether we want to stick with the "built-in COM support" term, or use the term @AaronRobinsonMSFT suggests (`ComInterop`), I would name this AppContext setting either:

* `System.Runtime.InteropServices.BuiltInCom.IsSupported`
    or
* `System.Runtime.InteropServices.ComInterop.IsSupported`

The MSBuild property name should then follow from this decision and match the above term:

* `BuiltInComSupport`
     or
* `ComInteropSupport`

Reply:
Thanks @eerhardt! I'll go with System.Runtime.InteropServices.BuiltInCom.IsSupported

Reply:
My 2 cents
* It would be really good if we ever exposed this as a public API (like we do for some of the other FS, or we plan to anyway) that the string matches the actual API name. I think both strings proposed by @eerhardt could work that way.
* Reading some of the other discussions, I tent to like `ComInterop` better. There's something about "built-in" which doesn't sound right (not sure what though).

Reply:
After chatting on this with folks, we settled on BuiltInComInteropSupport to continue on the precedence we set in the ComWrappers [doc](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.comwrappers). We want developers to move away from built-in COM interoperability support to ComWrappers for trim, AOT and other reasons.

AppContext Setting: `System.Runtime.InteropServices.BuiltInComInterop.IsSupported`
MSBuild property name: `BuiltInComInteropSupport`